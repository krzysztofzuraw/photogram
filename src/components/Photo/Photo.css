@font-face {
  font-family: "Merriweather";
  src: local("Merriweather");
}

.photo {
  box-shadow: 0px 2px 6px rgba(0, 0, 0, 0.18);
  background-color: #fff;
  font-family: "Merriweather", sans-serif;
  display: grid;
  grid-template-columns: auto;
  grid-template-rows: 50px auto 50px;
}

.photo-header {
  grid-row: 1 / 2;
  font-size: 1.3rem;
  display: flex;
  align-items: center;
}

.user-icon {
  fill: #1cba77;
}

.photo-header .user-icon {
  margin-left: 20px;
}

.photo-author {
  margin-left: 10px;
  text-decoration: none;
  color: #222;
}

.photo-author:hover {
  color: #1cba77;
}

.photo-img {
  grid-row: 3 / 2;
}

.photo-img:hover {
  cursor: pointer;
}

.photo-meta {
  grid-row: 3 / 3;
  display: flex;
  align-items: center;
  color: #586368;
}

.photo-hearth:hover {
  fill: #990000;
  transition: 200ms;
  transition-timing-function: ease-in-out;
  transform: translateY(-2px);
}

.photo-meta .photo-hearth {
  margin-left: 20px;
  margin-right: 10px;
}

.photo-meta .comments-icon {
  margin-left: 20px;
  margin-right: 10px;
}

.photo:hover {
  transform: translateY(-1px);
  box-shadow: 0 7px 14px rgba(50, 50, 93, 0.1), 0 3px 6px rgba(0, 0, 0, 0.08);
}
