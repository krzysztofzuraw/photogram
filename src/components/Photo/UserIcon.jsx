import React from "react";
import PropTypes from "prop-types";

const UserIcon = props => {
  const { width, height } = props;
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 20 20"
      width={width}
      height={height}
      className="user-icon"
    >
      <path d="M10 20a10 10 0 1 1 0-20 10 10 0 0 1 0 20zM7 6v2a3 3 0 1 0 6 0V6a3 3 0 1 0-6 0zm-3.65 8.44a8 8 0 0 0 13.3 0 15.94 15.94 0 0 0-13.3 0z" />
    </svg>
  );
};

UserIcon.PropTypes = {
  width: PropTypes.number,
  height: PropTypes.number
};

export default UserIcon;
