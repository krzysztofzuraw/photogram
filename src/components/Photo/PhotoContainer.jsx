import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Photo from "./Photo";
import { addLike } from "../../actions";

export class PhotoContainer extends React.Component {
  handleLike = () => {
    const { addLike, id } = this.props;
    addLike(id);
  };

  render() {
    return <Photo {...this.props} handleLike={this.handleLike} />;
  }
}

PhotoContainer.PropTypes = {
  addLike: PropTypes.func,
  id: PropTypes.string
};

const mapStateToProps = (state, ownProps) => ({
  commentsSum: state.rootReducer.comments.filter(
    comment => comment.photoId === ownProps.id
  ).length,
  likesSum: state.rootReducer.likes.filter(id => id === ownProps.id).length
});

export default connect(mapStateToProps, { addLike })(PhotoContainer);
