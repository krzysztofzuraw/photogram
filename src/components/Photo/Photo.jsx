import React from "react";
import { Link } from "react-router-dom";

import "./Photo.css";

import UserIcon from "./UserIcon";
import HearthIcon from "./HearthIcon";
import ConversationIcon from "./ConversationIcon";

class Photo extends React.Component {
  constructor(props) {
    super(props);

    this.state = { likes: 0 };
  }

  handleLike = () => {
    this.setState({ likes: this.state.likes + 1 });
  };

  render() {
    const { id, url, user: { name, links: { self } }, style } = this.props;
    return (
      <div className="photo" style={style}>
        <div className="photo-header">
          <UserIcon width={30} height={30} />
          <a href={self} className="photo-author">
            {name}
          </a>
        </div>
        <Link to={`/photo/${id}`}>
          <img alt="placeholder" className="photo-img" src={url} />
        </Link>
        <div className="photo-meta">
          <HearthIcon width={30} height={30} onClickHandler={this.handleLike} />
          <span>{this.state.likes}</span>
          <ConversationIcon width={30} height={30} />
          <span>20</span>
        </div>
      </div>
    );
  }
}

export default Photo;
