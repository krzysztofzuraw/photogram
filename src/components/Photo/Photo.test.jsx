import React from "react";

import { shallow } from "enzyme";

import Photo from "./Photo";

describe("Photo Component", () => {
  it("renders correctly", () => {
    const props = {
      id: "1",
      url: "https",
      user: {
        name: "Tom",
        links: {
          self: "https"
        }
      },
      style: {},
      commentsSum: 0,
      likesSum: 0,
      handleLike: jest.fn()
    };
    const wrapper = shallow(<Photo {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
