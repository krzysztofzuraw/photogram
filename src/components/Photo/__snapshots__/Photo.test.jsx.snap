// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Photo
    commentsSum={0}
    handleLike={[Function]}
    id="1"
    likesSum={0}
    style={Object {}}
    url="https"
    user={
        Object {
            "links": Object {
              "self": "https",
            },
            "name": "Tom",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="photo-header"
>
          <UserIcon
                    height={30}
                    width={30}
          />
          <a
                    className="photo-author"
                    href="https"
          >
                    Tom
          </a>
</div>,
        <Link
          replace={false}
          to="/photo/1"
>
          <img
                    alt="placeholder"
                    className="photo-img"
                    src="https"
          />
</Link>,
        <div
          className="photo-meta"
>
          <HearthIcon
                    height={30}
                    onClickHandler={[Function]}
                    width={30}
          />
          <span />
          <ConversationIcon
                    height={30}
                    width={30}
          />
          <span />
</div>,
      ],
      "className": "photo",
      "style": Object {},
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <UserIcon
              height={30}
              width={30}
/>,
            <a
              className="photo-author"
              href="https"
>
              Tom
</a>,
          ],
          "className": "photo-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 30,
              "width": 30,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tom",
              "className": "photo-author",
              "href": "https",
            },
            "ref": null,
            "rendered": "Tom",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="placeholder"
            className="photo-img"
            src="https"
/>,
          "replace": false,
          "to": "/photo/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "placeholder",
            "className": "photo-img",
            "src": "https",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HearthIcon
              height={30}
              onClickHandler={[Function]}
              width={30}
/>,
            <span />,
            <ConversationIcon
              height={30}
              width={30}
/>,
            <span />,
          ],
          "className": "photo-meta",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 30,
              "onClickHandler": [Function],
              "width": 30,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "height": 30,
              "width": 30,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="photo-header"
>
            <UserIcon
                        height={30}
                        width={30}
            />
            <a
                        className="photo-author"
                        href="https"
            >
                        Tom
            </a>
</div>,
          <Link
            replace={false}
            to="/photo/1"
>
            <img
                        alt="placeholder"
                        className="photo-img"
                        src="https"
            />
</Link>,
          <div
            className="photo-meta"
>
            <HearthIcon
                        height={30}
                        onClickHandler={[Function]}
                        width={30}
            />
            <span />
            <ConversationIcon
                        height={30}
                        width={30}
            />
            <span />
</div>,
        ],
        "className": "photo",
        "style": Object {},
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <UserIcon
                height={30}
                width={30}
/>,
              <a
                className="photo-author"
                href="https"
>
                Tom
</a>,
            ],
            "className": "photo-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "height": 30,
                "width": 30,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tom",
                "className": "photo-author",
                "href": "https",
              },
              "ref": null,
              "rendered": "Tom",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="placeholder"
              className="photo-img"
              src="https"
/>,
            "replace": false,
            "to": "/photo/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "placeholder",
              "className": "photo-img",
              "src": "https",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HearthIcon
                height={30}
                onClickHandler={[Function]}
                width={30}
/>,
              <span />,
              <ConversationIcon
                height={30}
                width={30}
/>,
              <span />,
            ],
            "className": "photo-meta",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "height": 30,
                "onClickHandler": [Function],
                "width": 30,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "height": 30,
                "width": 30,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
